app/code/Magento/CatalogInventory/Model/StockStateProvider.php

```diff
@@ -106,6 +106,10 @@ public function checkQuoteItemQty(StockItemInterface $stockItem, $qty, $summaryQ
     {
         $result = $this->objectFactory->create();
         $result->setHasError(false);
+        
+        $product = $this->productFactory->create();
+        $product->load($stockItem->getProductId());
+        $stockItem->setProduct($product);
```

Comment:
Also, there is unit test failure:
```
Fatal error: Call to undefined method Mock_StockItemInterface_837dff1c::setProduct() in /home/travis/build/magento/magento2/app/code/Magento/CatalogInventory/Model/StockStateProvider.php on line 112
```

Is there a way to achieve the same using only defined interfaces?